pandas/core/internals/blocks.py

```diff
@@ -1860,6 +1863,12 @@ def interpolate(
             placement=self.mgr_locs,
         )
 
+    def diff(self, n: int, axis: int = 1) -> List["Block"]:
```

Comment:
Note this is on ExtensionBlock now.  The deprecation for EAs not implementing `__sub__` required a similar block / axis handling, so I've moved it up here. The `_block_shape` is all the way up in Block.diff.